### PR TITLE
feat: allow eslint-plugin-promise 6

### PR DESCRIPTION
**Pull request by nschonni**
Expand the peerDependency and add a CI matrix to test it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "eslint": "^7.12.1",
     "eslint-plugin-import": "^2.22.1",
     "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-promise": "^4.2.1 || ^5.0.0"
+    "eslint-plugin-promise": "^4.2.1 || ^5.0.0 || ^6.0.0"
```
Comment (nschonni) [inline package.json:53]: Alternatly, this could just change to `>=4.21`, keep the testing, and update the matrix

**Comment by voxpelli** [comment]
Would be helpful with an explanation of why this new major version isn't breaking from our perspective.

**Comment by nschonni** [comment]
The v6 is more about ESLint 8 compat, but I added the CI testing to ensure that there is not conflicts going forward, since you seem to have set v4 as a baseline

**Comment by voxpelli** [comment]
Right. Looks like it's the node version that's the breaking change

**Comment by nschonni** [comment]
I can add the Node version matrix as well here if you want

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,6 +8,13 @@ on:
 
 jobs:
   ci:
+    strategy:
+      fail-fast: false
+      matrix:
+        eslint-plugin-promise:
```
Comment (voxpelli) [inline .github/workflows/ci.yml:14]: I would prefer to have this in something like a `compat.yml`, to keep the main CI file as lean as possible

File: package.json (hunk 1 above)
Comment (voxpelli) [reply package.json:53]: I like specifying the upper version that complies with the standard rule set, so an upgrade doesn't cause unintentional rule changes

**Comment by voxpelli** [comment]
> I can add the Node version matrix as well here if you want

Not necessary I think

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nschonni) [reply .github/workflows/ci.yml:14]: Extra YML files here would be interpreted as malformed workflow files. Might be something for when you go for the org-wide action approach and need a repo specific override file

**Review by Divlo — CHANGES_REQUESTED** [review]
Hey! :smile:

Thanks for your PR, but this will already be fixed by this PR: https://github.com/standard/eslint-config-standard/pull/193 that hopefully, we will soon merge and release, it is ready, I'm waiting for reviews.

**Comment by nschonni** [comment]
@Divlo yeah, I'm aware of the other PR, but it seems stalled because of the multiple changes. I opened this to fix one of the issues the other PR was dealing with that should be easier to land

**Comment by Divlo** [comment]
The PR #193 has been landed and released in the prerelease `17.0.0-0`.

**Pull request closed without merging**